src/threadpool_scheduler.cpp

```diff
@@ -64,15 +69,77 @@ struct threadpool_data {
 	std::condition_variable shutdown_complete_event;
 #endif
 };
-
+    
+#if defined(EMULATE_PTHREAD_THREAD_LOCAL)
+static pthread_key_t owning_threadpool_key;
+static pthread_key_t thread_id_key;
```

Comment:
Also wrap the key static in a function, as I explained above for `thread_wait_handler`.